lib/test_prof/recipes/rspec/let_it_be.rb

```diff
@@ -114,6 +113,53 @@ def define_let_it_be_methods(identifier, **modifiers)
 
       let(identifier, &let_accessor)
     end
+
+    # Exception needs to be handled both here and in `handle_frozen_hash_error`
+    # because if it is raised in before_all it isn't caught in `after` block and
+    # if it's inside the example it isn't raised so it has to be handled in `after`.
+    def build_let_it_be_initializer(identifier, freeze, &block)
+      proc do
+        begin
+          record = instance_exec(&block)
+
+          instance_variable_set(:"#{TestProf::LetItBe::PREFIX}#{identifier}", record)
+        rescue => e
+          e.message << FROZEN_ERROR_HINT if e.message.match?(/can't modify frozen/)
+          raise e
+        end
+      end
+    end
+
+    module Freezer
+      # Rerucsively freezes the object to detect modifications.
+      def self.deep_freeze(record, stoplist)
+        return if record.frozen?
+        return if stoplist.include?(record)
+
+        record.freeze
+
+        return record.each { |rec| deep_freeze(rec, stoplist) } if record.respond_to?(:each)
+
+        # Freeze associations as well.
+        #
+        # NOTE: `reload` statements in test or production code will cause
+        # a `FrozenError`. In case the use of `reload` cannot be avoided, use
+        # `reload: true` in `let_it_be` declaration.
+        return unless defined?(::ActiveRecord)
```

Comment:
Let's make it more specific (some gems could define `ActiveRecord` module without ActiveRecord itself to load their extensions):

```suggestion
        return unless defined?(::ActiveRecord::Base)
```